### PR TITLE
Bug fixes and update to Profile page

### PR DESCRIPTION
**Pull request by Thomas-Gallant**
Changelog:
- Creating a "Profile Toggle" component to reduce code duplication
- Setting tabsize to 2 in vscode settings
- Removing dynamic resizing of profile component and replacing with static sizes. A constant font size should be readable on all platforms
- Created a master "Highlight" toggle switch for controlling all highlight options. Created playwright tests for testing functionality
- Fixed issue with profile components not fitting on mobile devices
- Refactoring changelog to be in "past tense" to maintain consistency.

Before and After screenshots:

image

image

## Checklist for completing pull request:
- [x] Update Changelog.json if any functionality has been changed for the end user.
- [x] Test functionality on Mobile, Web, and Desktop
- [x] Verify that any tests for new functionality are created and functional.
- [ ] If needed, update the Readme

## Summary by CodeRabbit

- **New Features**
  - Introduced a new `ProfileToggle` component for user profile settings.
  - Added a new version entry in the changelog for version "1.24.1" with updates to the profile page and a highlight settings master switch.
  
- **Bug Fixes**
  - Resolved mobile compatibility issues related to the profile page.

- **Documentation**
  - Updated changelog entries for clarity and consistency.

- **Refactor**
  - Simplified `ProfilePage` component layout and logic, utilizing the new `ProfileToggle` for rendering settings. 

- **Tests**
  - Added end-to-end tests for the profile page's theme and highlight functionalities.

### CODE REVIEW

**Comment by Thomas-Gallant** [comment]
@coderabbitai review

File: app/Components/Profile/ProfileToggle.tsx [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+  name: string;
+  fontSize?: number;
+  value: boolean;
+  valueToggle: (() => void) | ((mode: boolean) => void);
+  testIdPrefix: string;
+}) => {
+  let fontSize = props.fontSize;
+  if (!props.fontSize) {
+    fontSize = 25;
+  }
```
[the remark this answers is not in the document]
Comment (Thomas-Gallant) [reply app/Components/Profile/ProfileToggle.tsx]: This has been updated, thanks!

File: Changelog.json [hunk 2]
```diff
@@ -344,7 +355,7 @@
     "date": "August 19th, 2023",
     "summary": "Sudoku games no longer rely on user needing Internet access.",
     "features": [
-      "Staring a Sudoku game now retrieves a puzzle from the user's device instead of requring Internet access.",
+      "Starting a Sudoku game now retrieves a puzzle from the user's device instead of requring Internet access.",
       "Games are saved locally on the user's device instead of requring an Internet connection to save games.",
```
[the remark this answers is not in the document]
Comment (Thomas-Gallant) [reply Changelog.json]: This has been fixed, thanks!

File: Changelog.json [hunk 3]
```diff
@@ -55,7 +68,7 @@
   {
     "version": "1.21.1",
     "date": "July 18th, 2024",
-    "summary": "Fixing issue where some pages in app overlap with phone icons on top of page (time, battery indicator, etc).",
+    "summary": "Fixed issue where some pages in app overlap with phone icons on top of page (time, battery indicator, etc).",
     "bug fixes": [
```
[the remark this answers is not in the document]
Comment (Thomas-Gallant) [reply Changelog.json]: This has been fixed, thanks!

File: Changelog.json [hunk 4]
```diff
@@ -393,44 +404,44 @@
...
     "targets": ["web", "mobile"],
     "contributors": ["Gregory711"]
   },
   {
     "version": "0.34.9",
     "date": "April 17th, 2023",
-    "summary": "Adding back selected font for Sudoku board.",
-    "bug fixes": ["Adding back selected for for Sudoku board."],
+    "summary": "Added back selected font for Sudoku board.",
+    "bug fixes": ["Added back selected for for Sudoku board."],
```
[the remark this answers is not in the document]
Comment (Thomas-Gallant) [reply Changelog.json]: This has been fixed, thanks!

File: Changelog.json [hunk 5]
```diff
@@ -276,27 +289,25 @@
...
     "contributors": ["Thomas-Gallant"]
   },
   {
     "version": "1.8.1",
     "date": "August 31st, 2023",
-    "summary": "Fixing bug with inputing values into Sudoku board",
+    "summary": "Fixed bug with inputing values into Sudoku board",
     "bug fixes": [
-      "Fixing an issue where the demo board was sharing state with the play board, causing inputs to the play board to be disabled when the demo board was in 'hint' mode."
+      "Fixed an issue where the demo board was sharing state with the play board, causing inputs to the play board to be disabled when the demo board was in 'hint' mode."
```
[the remark this answers is not in the document]
Comment (Thomas-Gallant) [reply Changelog.json]: This has been fixed, thanks!

**Review by Gregory711 — APPROVED** [review]

**Pull request merged**